### PR TITLE
Improve modeling of `Child` in CodeGeneration

### PR DESCRIPTION
**Pull request by ahoppen**
Two changes:
- Make `Child.varName` a `TokenSyntax`
- Change type of `Child.documentation` to `Trivia`
  - This also fixes a bug that caused multi-line documentation of children to be rendered incorrectly

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: CodeGeneration/Sources/SyntaxSupport/Child.swift [hunk 1]
```diff
@@ -56,15 +58,41 @@ public enum ChildKind {
...
+  ///
+  /// This is used to e.g. describe the child if all of its tokens are missing in the source file.
   public let nameForDiagnostics: String?
-  public let documentation: String?
+
+  /// A doc comment describing the child.
+  public let documentation: SwiftSyntax.Trivia
+
+  /// The first line of the child's documentation
+  public let documentationAbstract: String
```
[the remark this answers is not in the document]
Comment (ahoppen) [reply CodeGeneration/Sources/SyntaxSupport/Child.swift:86]: Yes, the problem is that we use this string for the parameter documentation of the node’s initializers. And for those, we don’t want it prefixed with `///`

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by kimdv — APPROVED** [review]

**Review by Andrii3664 — APPROVED** [review]

**Pull request merged**